dev/service-registry/client-javadsl/src/main/java/com/lightbend/lagom/internal/javadsl/registry/ServiceRegistryService.java

```diff
@@ -7,34 +7,43 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 
 import java.net.URI;
-import java.util.List;
-import java.util.Optional;
+import java.util.*;
 import java.util.stream.Collectors;
 
 /**
  * A service to be registered by the service registry
  */
 public class ServiceRegistryService {
-    private final URI uri;
+    private final List<URI> uris;
```

Comment:
This change is quite important. In dev mode, when a new service (unmanaged or managed) is registered into the service registry, an HTTP request with this payload is sent.

With the change here, a single service can now register many URIs. The `scheme` information on the `uris` will be used on the service registry to setup the `portName`.

Reply:
Maybe a `Map<String, URI>` would be better. Other than being more explicit, it allows for the possibility of multiple http or multiple https ports (which could be used for the Akka Management HTTP server for example).

Reply:
Another consideration: most real world service registries don't have a notion of a "scheme" and don't return complete URIs, but only host:port pairs. Similarly, most clients only care about the host and the port (e.g., Cassandra and Kafka) and ignore the scheme. The fact that Lagom's `ServiceLocator` API works in terms of URIs is an unusual quirk that I think we should move away from as part of the migration to Akka Discovery.

Reply:
I tried `Map<String, URI>` but required either (1) duplicating the logic to extract the `portName` from the scheme in both `ServiceRegistryService` (java and scala) or (2) require the `portName` in every callsite of `ServiceRegistryService`. 

Also, supporting multiple `http` ports in a single `serviceName` (in a single process) is not possible since it creates ambiguous lookups. Either you use a different service name or a different `portName`, but using `inventoryService:http` would be unclear and, instead, I'd probably have to register `inventoryService:http` and `inventoryService-management:http`. WDYT?

About the second comment, I agree that we must move away from URIs in both the registration and lookup. Passing the scheme information will become unnecessary when `portName` is added in registration and assumed in lookup.

Reply:
Moving this `URI` to `InetAddress` should eventually cause a change in the API of `unmanagedServices` in the build tool settings so users are required to provide a `portName` and an `InetAddress` instead of an URI from which we extract the details.